Initial set up of the postBase set

llrs opened the pull request:
Here it is my post on BaseSet. 
@stefaniebutland let me know of further improvements and changes (style, tone...). 

* [x] I have read the Content Guidelines.
* [x] I have read the Technical Guidelines.
* [x] I used or followed the R Markdown or Markdown template.
* [x] I have followed the Style Guide.
* [x] I created or updated my author metadata with correct folder name.
* [x] I have added relevant tags after browsing existing tags.
* [x] I ran `roblog::ro_lint_md()` on index.md (optional).
* [x] I ran `roblog::ro_check_urls()` on index.md (optional).
 There's a warning because I used the doi: https://doi.org/10.1371/journal.pcbi.1002375 which gets redirected.
* [x] I ran a spell-check on index.md.
* [x] I have added the tags - Software Peer Review, my-packagename, R.
* [x] I have added the package-version YAML tag.
* [x] I have added acknowledgement of the reviewers' work.
* [x] I have added a link to the software peer review thread.

stefaniebutland commented:
Thank you @llrs! My colleague @steffilazerte will review the post. Tentative publication dates 2021-01-19 or 2021-01-26.

Steffi fyi just for some background, Lluís has published a posts about the review processes with Bioconductor and rOpenSci
- https://llrs.dev/2020/06/bioconductor-submissions/
- https://llrs.dev/2020/07/bioconductor-submissions-reviews/
- https://llrs.dev/2020/09/ropensci-submissions/

steffilazerte commented:
Hi @llrs 👋🏼 ! 

I'll do a complete review on Monday, but in the meanwhile, could you compile (knit) the index.Rmd file and add the index.md to the pull request? Leave the index.Rmd in the pull request as well. We need the index.md to see the live preview.

Once checks have passed, you can click "Details" to the right of "Deploy preview ready!" in the Checks section below.

llrs commented:
I thought the *.md would be created on the fly by Netlify, I added it now. There's a mismatch between my local version of Hugo and the one on Netlify, as discovered with `blogdown::check_site()`
```
 blogdown is using Hugo version (0.67.1) to build site locally.
  Netlify is using Hugo version (0.73.0) to build site.
```

I think it shouldn't affect, but just so you know.

steffilazerte commented on content/blog/2021-01-14-introducing-baseset/index.Rmd:
```diff
@@ -0,0 +1,230 @@
+  - BaseSet
+  - Software Peer Review
+  - data-munging
+  - R
```
We're just in the process of changing this now to remove `R` as a suggested tag :smile:
```suggestion
```

steffilazerte reviewed (commented):
Hi @llrs, 
Thanks for submitting your blog post!

The package is interesting and I really like the background story of how BaseSet came to be and the gentle introduction to those of us (me!) unfamiliar with more advance base set mathematics. 

I have 48 suggestions in the pull request, while that may seem like a lot, I have one or two suggestions for a bit more detail, but on the whole my comments are mostly just minor wording suggestions.

With the exception of the those highlighted below in the checklist, feel free to accept, reject or modify any suggestions I've made!

For publication dates, we have Tuesday Jan 19th or Tuesday Jan 26th. If you'd like next Tuesday (the 19th), could you get your revisions to me by Friday Jan 15th? 
Otherwise, if you're aiming for Jan 26th, the revisions can wait until Friday Jan 22nd.

Thanks!

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [x] publication date is ok 
  - We have Tues Jan 19th or Tues Jan 26th as publication date options which would you prefer?
* [X] alternative text of images is informative
* [X] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
* [ ] author metadata is provided with correct folder name
  - Make sure that the folder name is **exactly** the same as your name (i.e. change the i to í)
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on index.md
* [X] I ran `roblog::ro_check_urls()` on index.md
* [X] I ran a spell-check on index.md
* [X] YAML subject tags are ok
* [X] YAML package-version included
* [X] YAML subject tags - software peer review, packagename; add "community" for post by non-staff non-editor
* [ ] acknowledges and links to reviewers
  - I've suggested links to reviewers in the text
* [X] links to peer review thread

llrs commented:
I tried to catch the reader with a history first. Glad you liked the introduction of the set theory, I wasn't sure it would be gentle enough. Many thanks for the detailed review, apologies for the many mistakes. Hope it is ready for the 19th. 

I adopted all your suggestions about wording and text of the post. I added a link to the third package that I didn't mention previously and provided the operations behind the numbers that appear on the text. I tried to show a formula with latex but a) I didn't manage b) don't think it is the right place (wouldn't like to scare possible users). Also modified some few sentences at the end of the post.
 
The technicalities of the website (date, twitter handle, name of the folder...) are fixed too. By the way, I used the new feature of the upcoming RStudio version named markdown visual editor to write the post. The menu converted the text I wrote as hover text as that markdown notation you saw (Neither did I know how to do that!).

llrs commented:
@steffilazerte Missed one change. Hope I didn't miss anything else (should have used the "Add suggestion to batch" :sweat_smile: ) and this is ready for publication. If you have more suggestions let me know!

steffilazerte commented on content/blog/2021-01-19-introducing-baseset/index.Rmd:
```diff
@@ -0,0 +1,234 @@
+So I got in touch with them and we shared some ideas on how to do this.
+
+To explore different ways to solve the problem we ended up developing three different packages that aim to do similar things but take different approaches.
+At the moment BaseSet published on CRAN, [BiocSet](https://bioconductor.org/packages/BiocSet/ "BiocSet displays different biological sets in a triple tibble format") is published on Bioconductor and [unisets](https://github.com/kevinrue/unisets) that the main developer can't dedicate more time.
```
Perhaps a more gentle wording suggestion?
```suggestion
At the moment BaseSet published on CRAN and [BiocSet](https://bioconductor.org/packages/BiocSet/ "BiocSet displays different biological sets in a triple tibble format") is published on Bioconductor.  [unisets](https://github.com/kevinrue/unisets) is available on GitHub although development has stalled due to time constraints.
```

llrs replied:
Thanks, I struggled with that one!

steffilazerte commented on content/blog/2021-01-19-introducing-baseset/index.Rmd:
```diff
@@ -0,0 +1,234 @@
+> [Categories](https://en.wikipedia.org/wiki/Help:Category "Help:Category"): [Basketball](https://en.wikipedia.org/wiki/Category:Basketball "Category:Basketball") \| [Sports originating in the United States](https://en.wikipedia.org/wiki/Category:Sports_originating_in_the_United_States "Category:Sports originating in the United States") \| [Team sports](https://en.wikipedia.org/wiki/Category:Team ...
+
+Some say that tomatoes should be both considered as fruits and as vegetables, so they could also be in both sets.
+We can do this for all elements we can think, pencils, mathematical operations, numbers, words, genes...
```
```suggestion
We can do this for any element we can think of, pencils, mathematical operations, numbers, words, genes...
```

steffilazerte commented on content/blog/2021-01-19-introducing-baseset/index.Rmd:
````diff
@@ -0,0 +1,234 @@
+element_size(TS)
+```
+
+Given this memberships there is just a 0.3252 ( `0.1*(1-0.2)*(1-0.6)*(1-0.85) + (1-0.1)*0.2*(1-0.6)*(1-0.85) + (1-0.1)*(1-0.2)*0.6*(1-0.85) + (1-0.1)*(1-0.2)*(1-0.6)*0.85`) probability that the machine would classify the first image as just one thing, so it is more probable that the first image will be misclassified. 
````
Eeep! This is more complex than I expected! I can see why you didn't include the formula :grin: I think still it's important to show readers where the numbers come from, but perhaps put this calculation and the next as footnotes?

llrs replied:
Done, added the calculation as footnotes with a very brief explanation.

steffilazerte reviewed (commented):
Hi @llrs 

Thanks for your speedy revisions. You've done so much work for us already, we really appreciate it! So I wouldn't think of the wording tweaks as errors, just as my job and contribution as the editor. 

Thanks for fixing the dates, author folder etc. I have 3 final suggestions, but other than that we're good to go for Tuesday Jan 19th 2021! 

I'll post a draft tweet here, which we'll use to advertise your post on Tuesday. Feel free to comment on it if you have suggestions.

Thanks again!

llrs commented:
Many thanks for the edition and tweaks! I greatly appreciate them, as this year I'm focusing on improving my communication skills :mega:

steffilazerte commented:
One last thing (I promise!) I was just finalizing the tweet draft (see below) and it occurred to me that the title should be a bit more informative. 

Do you have an idea for a more informative title?

If you're unsure, here are some potential options:

"Introducing BaseSet for mathematical sets applied to genes"
"Introducing BaseSet for mathematical sets"
"Introducing BaseSet for exploring gene sets"
"Using BaseSet to explore gene sets"
"BaseSet extends mathematical sets in R"

You've done so much work already, if you just let me know here what title you'd like, I can make the changes to the pull request and have it ready for Tuesday 😄 

---

Tweet Draft (I'll update the title later):

[blog] Introducing BaseSet 🧬

@Lluis_Revilla shares the story behind BaseSet, a package for comparing mathematical sets.
🔵 🟢 🟥 🟩

Check it out to find out how this can help with the study of genes 🧬 related to disease!

https://ropensci.org/blog/2021/01/19/introducing-baseset/ 

#rstats #BaseSet

llrs commented:
I think the best title would be "Introducing BaseSet for mathematical sets". The package can be used for more things than genes. 
Nice tweet, I like the emojis used

steffilazerte commented:
Great!
